ArduPlane/AP_Arming.cpp

```diff
@@ -387,6 +398,26 @@ void AP_Arming_Plane::update_soft_armed()
 
         delay_arming = false;
     }
+
+#if AP_PLANE_BLACKBOX_LOGGING
+    if (blackbox_speed > 0) {
+        const float speed3d = plane.gps.status() >= AP_GPS::GPS_OK_FIX_3D?plane.gps.velocity().length():0;
+        const uint32_t now = AP_HAL::millis();
+        if (speed3d > blackbox_speed) {
+            last_over_3dspeed_ms = now;
+        }
+        if (!_armed && speed3d > blackbox_speed) {
+            // force safety on so we don't run motors
+            hal.rcout->force_safety_on();
+            arm(Method::BLACKBOX, false);
```

Comment:
I suspect this doesn't work anymore now we have a mandatory RC check, or at least there is some more setup required.

Reply:
good point, I'll test in SITL and add an autotest to make sure it keeps working